tests/core/test_eoexecutor.py

```diff
@@ -143,7 +142,11 @@ def test_read_logs(test_args, execution_names, workflow, execution_kwargs, logs_
         )
         executor.run(workers=workers, multiprocess=multiprocess)
 
-        execution_logs = executor.read_logs()
+        execution_logs = []
+        for log_path in executor.get_log_paths():
+            with open(log_path) as f:
+                execution_logs.append(f.read())
+
```

Comment:
Seems like the `read_logs` pops up quite often, are you sure you want to deprecate it?

Reply:
i see no reason why the executor would read logs for you. I even think that the current approach where log paths are slapped onto it is wrong. In the future we should make it stateless, and all of these things get pinned to a 'result' object.

the only place in the code where it pops up is generating the report, other uses are just in tests because it is convenient